WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -359,8 +361,15 @@ def my_competitions
       competition_params[:eventSpecs] = competition_params[:event_ids].select { |k, v| v == "1" }.keys.join " "
       competition_params.delete(:event_ids)
     end
-    if params[:commit] == "Confirm" && current_user.can_confirm_competition?(@competition)
-      competition_params[:isConfirmed] = true
+    # Ensure that the user is a delegate if they are confirming the competition.
+    if params[:commit] == "Confirm"
+      if @competition
+        if current_user.can_confirm_competition?(@competition)
+          competition_params[:isConfirmed] = true
+        end
+      elsif competition_params[:delegate_ids].split(",").include? current_user.id.to_s
```

Comment:
If we confirm a competition from the new page there is no @competition at this point in the code so in that case I'm checking that the user is in the delegate_ids to ensure they can confirm this competition.